app/models/folio/pickup_destination.rb

```diff
@@ -18,9 +18,9 @@ def display_label
     end
 
     # For paging scheduling, we must map from service point to library if folio
-    # Otherwise, the library code is what we will be receiving
+    # For service points which don't map to libraries in FOLIO, will rely on Settings
     def library_code
-      Folio::Types.instance.map_to_library_code(@code)
+      Folio::Types.instance.map_to_library_code(@code) || (Settings.libraries[@code].present? ? @code : nil)
```

Comment:
Haven't tested, but I think you can say `@code if Settings.libraries.key?(@code)` and it'll be `nil` on its own otherwise

Reply:
The '.key?(@code)' hadn't worked for me but I can try the if statement approach

Reply:
Updated.